services/statestorage/merkle/merkle_forest.go

```diff
@@ -0,0 +1,236 @@
+package merkle
+
+import (
+	"fmt"
+	"github.com/orbs-network/orbs-network-go/crypto/hash"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+type RootId uint64
+type Proof []*Node
+
+var zeroValueHash = hash.CalcSha256([]byte{})
+
+type Node struct {
+	path     string
```

Comment:
I'm not sure I understand how path works, would it be possible to add a comment on the path format? Also, why is it a string?

Reply:
temporary implementation. path will eventually be a []byte with an additional parity indication. This too is pending final spec updates.

Currently using simple string as substitute. This should not reach production as is. adding //TODO